Helper functions for the repeater fields

lukecarbis opened the pull request:
This PR adds the following helper functions:

- `block_rows()`
- `block_row()`
- `block_sub_field()`
- `block_sub_value()`

lukecarbis committed:
Remove maybe_get_sub_field_name helper function

lukecarbis committed:
Stub out helper functions for sub fields

lukecarbis committed:
Add Loop class and loop helper functions

lukecarbis committed:
Move typecasting from block_field to the Field class

lukecarbis committed:
Refactor block_field to resolve undefined index bugs

lukecarbis committed:
Add active property to loop class

lukecarbis committed:
First pass at block_sub_field

lukecarbis committed:
Minor docblock ammendment

lukecarbis committed:
Merge branch 'feature/repeater' into feature/repeater-helpers

lukecarbis committed:
Merge branch 'feature/repeater' into feature/repeater-helpers

lukecarbis committed:
Update sub field helper functions to match repeater data storage

lukecarbis commented:
@kienstra Ready for you to review.

kienstra commented on php/blocks/class-loop.php:
```diff
@@ -0,0 +1,92 @@
+	 *
+	 * @param string $name The field name.
+	 */
+	public function active( $name ) {
```
OK, nice. It looks like only one row can be active at a time, though there might be a few rows stored in `$this->loops`.

kienstra commented on php/blocks/class-loop.php:
```diff
@@ -0,0 +1,92 @@
+/**
+ * Class Loop
+ */
+class Loop {
```
It's nice how this clas follows the [Single Responsibility Principle](https://en.wikipedia.org/wiki/Single_responsibility_principle) really well. It only handles the loop, and doesn't have to know about what it's looping through.

kienstra committed:
Minor updates to the Loop class

Also, add a cast helper method.
Simply the previous logic,
extracted to a helper method.

kienstra committed:
Update the DocBlock summary of cast_value_to_string()

I should have changed this in
the previous commit, where the function
name changed.

kienstra reviewed (approved):
**Approved**

Hi @lukecarbis,
Good work! This looks good, and will help a lot with the progress on the `feature/repeater` branch.

kienstra commented:
For reference, here's how to use these repeater helper functions in a template:

```php
$repeater_name = 'my-repeater';
if ( block_rows( $repeater_name ) ) :
	while ( block_rows( $repeater_name ) ) :
		block_row( $repeater_name );
		
		block_sub_field( 'your-example-field' );
	endwhile;
endif;

```
